test/addons-napi/test_threadsafe_function/binding.c

```diff
@@ -0,0 +1,221 @@
+#include <uv.h>
```

Comment:
is libuv ABI stable? I _think_ there is going to be a libuv@2 sooner rather than later (cc @jasnel @cjihrig). For this example to work across runtimes, `uv.h` would need to be ABI stable as well.

I think we might have to add some wrappers around `uv.h` as well, but I might be mistaken.

Reply:
@mcollina I'm not sure how to test this without spawning threads. In the worst case I could copy in the relevant parts of libuv (`uv_thread_create()` and `uv_thread_join()`).

Reply:
@mcollina FWIW I haven't seen any [LEPs](https://github.com/libuv/leps) regarding threads.

Reply:
@gabrielschulhof I'm not referring to this test specifically, I've commented here to show an example. Do you think that our users will `#include <uv.h>` to use this function or not?

Reply:
@mcollina OK, I see. The usual use case for this is that on the native side there is already an implementation that uses threads, the maintainers of which do not wish to migrate to `napi_async_work`, but do wish to call into JS. Therefore the thread implementation already in use may or may not be libuv. In fact, the essential feature is that we need not care what threading library produces the non-main-loop threads from which we receive the calls.

We may wish to advise users of the possibility of future ABI breakage in libuv when they consider a new multi-threaded native implementation. Nevertheless, such advice is IMO orthogonal to this PR.

Reply:
awesome thanks. Can you add a comment to the test, so that's clear? Some people look for the tests to see how something would work, and this would look confusing.

Reply:
@mcollina can do.

Reply:
@mcollina done.

Reply:
@mcollina I did https://github.com/libuv/libuv/pull/1657 specifically to make libuv usable in an ABI-stable fashion if that is desired.